src/pages/home/report/ReportView.js

```diff
@@ -7,37 +7,41 @@ import {addAction} from '../../../libs/actions/Report';
 import KeyboardSpacer from '../../../components/KeyboardSpacer';
 import styles from '../../../styles/styles';
 import SwipeableView from '../../../components/SwipeableView';
+import FullScreenLoadingIndicator from '../../../components/Loading/FullscreenLoading';
+import withDrawerState from '../../../components/withDrawerState';
 
 const propTypes = {
-    // The ID of the report actions will be created for
+    /* The ID of the report the selected report */
     reportID: PropTypes.number.isRequired,
-};
 
-// This is a PureComponent so that it only re-renders when the reportID changes or when the report changes from
-// active to inactive (or vice versa). This should greatly reduce how often comments are re-rendered.
-class ReportView extends React.Component {
-    shouldComponentUpdate(prevProps) {
-        return this.props.reportID !== prevProps.reportID;
-    }
+    /* Is the view ready to be displayed */
+    loaded: PropTypes.bool.isRequired,
+
+    /* Is the report view covered by the drawer */
+    isDrawerOpen: PropTypes.bool.isRequired,
+};
 
```

Comment:
This is no longer needed. And I don't think it brought up any optimizations as all the heavy lifting is performed in the <ReportActionsView />